src/services/staking_data_service.ts

```diff
@@ -0,0 +1,161 @@
+import * as _ from 'lodash';
+import { Connection } from 'typeorm';
+
+import { Epoch, Pool, PoolWithStats, RawEpoch, RawPool } from '../types';
+import { stakingUtils } from '../utils/staking_utils';
+import { utils } from '../utils/utils';
+
+export class StakingDataService {
+    private readonly _connection: Connection;
+    constructor(connection: Connection) {
+        this._connection = connection;
+    }
+    public async getCurrentEpochAsync(): Promise<Epoch> {
+        const rawEpoch: RawEpoch | undefined = _.head(await this._connection.query(`SELECT * FROM staking.current_epoch;`));
+        if (!rawEpoch) {
+            throw new Error('Could not find the current epoch.');
+        }
+        const epoch = stakingUtils.getEpochFromRaw(rawEpoch);
+        return epoch;
+    }
+    public async getNextEpochAsync(): Promise<Epoch> {
+        const rawEpoch: RawEpoch | undefined = _.head(await this._connection.query(`SELECT
```

Comment:
In general I think it's worth separating the queries from these blocks of code and refer to an object here.